metrics/timer.go

```diff
@@ -0,0 +1,28 @@
+package metrics
+
+import "time"
+
+// Timer acts as a stopwatch, sending observations to a wrapped histogram.
+// It's a bit of helpful syntax sugar for h.Observe(time.Since(x)).
+type Timer struct {
+	h Histogram
+	t time.Time
+}
+
+// NewTimer wraps the given histogram and marks the current time.
```

Comment:
I'd avoid the work "mark" as that's used by Dropwizard Meters and may cause confusion, 'records' or 'notes' should do.